### ISSUE TITLE
[SCREENREADER] Select Facility Type and Select Service Type controls reads out as buttons

### ISSUE THREAD

**Issue by peterhill-oddball** [department-of-veterans-affairs/va.gov-team#1056]
### Issue origin:
February 2018 smoke test - Facility and Service Locator

### Issue:
The Select Facility Type and Select Service Type controls are custom dropdowns that act like `listboxes`, but announce themselves as `buttons` to screen readers. This leaves screen reader users with an incomplete understanding of how to select an option. 

These are completely custom elements; standard `<select>` menus won't allow us to include the icons, so I am referring to the WAI-ARIA practices for a rich listbox for markup guidance: https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html

### Pages affected:
* https://staging.vets.gov/facilities/

### Possible fix:
The `<Select />` and `<Listbox />` components will both need to be refactored. The trigger will need to be changed to a `<button>` with an `aria-haspopup="listbox"` and an `aria-expanded="false"` attribute. The unordered list will need to be associated with the button using `aria-labelledby` and the button ID. 

Please see the WAI-ARIA link above for further ARIA attributes that should be applied to the button, unordered list, and list items.

### Acceptance criteria
As a user, I want the custom listboxes to respond in the following ways:
* Receive keyboard focus by pressing `Tab`
* Expand and collapse by pressing `Spacebar` once and twice respectively
* Expand and collapse by pressing `Enter` once and twice respectively
* Expand and collapse by clicking with a mouse once and twice respectively
* Highlight the active list item when I hover with the mouse, or use the up and down arrow keys 
* Jump to the first list item when I press the `Home` key
* Jump to the last list item when I press the `End` key
* Cycle through the list items when the listbox has focus, but is not expanded. The WAI-ARIA listbox example opens the menu when this action is initiated, but I think if we can cycle through list items with the listbox closed, that would be closer to a standard `<select>` menu behavior.
* Announce the selected list item to screen readers when cycling through using the arrow keys

**Comment by Karak888** [comment]
@peterhill-oddball  will put this into top of our backlog for prioritizing.  Thanks!

**Comment by ncksllvn** [comment]
After looking into this, the scope for this ticket is a bit more than expected. It looks like that the `Select a facility type` dropdown is using an external module, [Downshift](https://www.npmjs.com/package/downshift), under the hood. We're using a very old version. In the updated versions, it looks like accessibility support is improved substantially. 

I updated our Downshift version to receive the accessibility improvements in the Facility Locator, but unfortunately that module is used throughout other areas of the site. Our forms library is one area that I'm especially hesitant to touch. 

I'm wondering if we should this one out in favor of a new ticket - "[508] Upgrade the Downshift module to latest version".

Side note - The `Select service type` dropdown appears to be an ordinary select menu now, so it should be okay.

---

@1Copenut I'd like your input on this when you have time. The updates I made to the Facility Locator are [here](https://github.com/department-of-veterans-affairs/vets-website/pull/10829) if you want to run it.

**Comment by 1Copenut** [comment]
@ncksllvn I'm happy to dig into this one further. I've got a lot of meetings today, so I'll put my initial comment in while I've got a few.

The WAI-ARIA 1.1 specification ( https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html ) for comboboxes creates a lot of challenges for screen readers. I have previously upgraded an Autocomplete component to Downshift V3, but had to manually re-write it to use the better supported 1.0 specification ( https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.0pattern/combobox-autocomplete-list.html )

The 1.0 specification with listbox is well supported, and I'm hearing discussion that the 1.1 spec might be deprecated. I think the right answer is to update Downshift, but be prepared to refactor the component so it abides the 1.0 spec so we don't introduce a usability regression.

**Comment by Karak888** [comment]
Closing this ticket as LOE is greater than anticipated. Upgrading specific component and FL would receive those updates at later time.  @ncksllvn will open a new ticket for VSP.

**Comment by ncksllvn** [comment]
New ticket for updating Downshift site-wide is here, https://github.com/department-of-veterans-affairs/va.gov-team/issues/2418.
